use/2015-12-15-microsatellite-differentiation.Rmd

```diff
@@ -0,0 +1,284 @@
+bov_amova
+```
+If the population was panmictic, we would expect to see extremely small variance
+components for Country and Breed compared to Error (i.e. the variation from
+individuals within populations). We don't see that and further suspect moderate
+differentiation. To test for this, you can have boostrap replicates by setting
+the `nperm` argument. Again, because these bootstraps take a long time, we'll
+set it to 10:
+
+```{r pegas_amova_boot}
+set.seed(20151219)
+bov_amova <- pegas::amova(bov_dist ~ Country/Breed, data = bov_stra, nperm = 10)
+bov_amova
+```
+
+Now we have evidence that these are significantly different.
+
+### Other implementations
+
+It is possible to do the same analyses in different packages, which I will list
+here.
+
+The *vegan* package has the function `adonis()` that implements a multivariate
+AMOVA and takes the same kind of arguments as the *pegas* implementation of
+`amova()`:
+
+```{r adonis, eval = FALSE}
+library("vegan")
+set.seed(20151219)
+res <- adonis(bov_dist ~ Country/Breed, data = bov_stra, permutations = 99)
+```
+
+The package *ade4* implements the original model, but it requires an abundance 
+matrix of haplotypes per population. The advantage of this version is the fact 
+that it calculates $\phi$ statistics for you. The package *poppr* provides a 
+wrapper for the *ade4* version that will sanitize the data and allows for
+calculation of within sample variance.
+
+```{r, poppr.amova, eval = FALSE}
+library("poppr")
+res <- poppr.amova(microbov, ~Country/Breed, within = FALSE)
+set.seed(20151219)
+randtest(res, nrepet = 99)
+```
+
+## Discriminant Analyses of Principal Components (DAPC)
+
+Finally, we will use DAPC in order to assess which populations have significant 
+structure (Jombart et al., 2011). For this, we will use the function `dapc()`
+from *adegenet*. 
+
+> Warning: This analyis is sensitive to the number of principal components you
+choose, so by default, you should run this function with just the data set and
+without any options beforehand (`dapc(microbov)`).
+
+```{r dapc}
+setPop(microbov) <- ~Country/Breed
+bov_dapc <- dapc(microbov, n.pca = 100, n.da = 4) # 100 principal components; 4 discriminant functions
+scatter(bov_dapc)
+```
+
+From this output, we can see that there is a lot of differentiation based on
+geography, but also that African breeds are a lot more differentiated than
+European breeds.
+
+# Conclusions
+
+We have assessed differentiation of these cattle breeds using tools in R. From 
+the diversity statistics, we can clearly see that there is moderate 
+differentiation among breeds. This is supported and further bolstered by the 
+results of AMOVA, showing us that there is significant differentiation among 
+countries and breeds. The DAPC analysis gives us a clear picture showing highly 
+differentiated African breeds with low diversity in the French breeds. Further
+steps could be to further assess factors such as inbreeding coefficients.
```

Comment:
Change ' steps could be to further assess factors such as inbreeding coefficients.' to  'steps could be to assess factors such as inbreeding coefficients.'